server/src/main/java/org/elasticsearch/cluster/metadata/MetadataIndexTemplateService.java

```diff
@@ -505,17 +505,51 @@ public static void validateV2TemplateRequest(Metadata metadata, String name, Com
         }
 
         final Map<String, ComponentTemplate> componentTemplates = metadata.componentTemplates();
+        final List<String> ignoreMissingComponentTemplates = (
+            template.getIgnoreMissingComponentTemplates() == null? List.of():template.getIgnoreMissingComponentTemplates()
```

Comment:
I was going forth and back on this one. Either I check for null in `getIgnoreMissingComponentTemplates` and always return an empty list or do it here. It seems for non requirement params `null` is in many places and excepted value. Also it put a bit a challenge in many other places that expect the entry to not show up by default if it is entry. So I do the check here.